src/main/scala/rules/Evaluator.scala

```diff
@@ -1425,7 +1444,7 @@ object evaluator extends EvaluationRules {
         case `stop` => Q(s1, t0, v1) // Done, if last expression was true/false for or/and (optimisation)
         case _ =>
           joiner.join[Term, Term](s1, v1)((s2, v2, QB) =>
-            brancher.branch(s2, t0, v2, true) _ tupled swapIfAnd(
+            brancher.branch(s2, t0, exps.reduce((e1: Exp,e2: Exp) => SilAnd(e1,e2)(NoPosition,NoInfo,NoTrafos)), v2, true) _ tupled swapIfAnd(
```

Comment:
Can you use https://github.com/viperproject/silicon/blob/master/src/main/scala/Utils.scala#L115 instead of `exps.reduce`?